src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/NuGetBrokeredServiceFactory.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.VisualStudio.Threading;
+
+namespace NuGet.VisualStudio.Implementation.Extensibility
+{
+    public class NuGetBrokeredServiceFactory
+    {
+        private readonly JoinableTaskFactory _joinableTaskFactory;
+
+        public NuGetBrokeredServiceFactory(JoinableTaskFactory joinableTaskFactory)
+        {
+            _joinableTaskFactory = joinableTaskFactory ?? throw new ArgumentNullException(nameof(joinableTaskFactory));
```

Comment:
You dont use it here...I guess there's stuff that isn't pushed yet :)

Reply:
Good catch. Initially I was trying to use an `AsyncLazy` to get `IProjectSystemCache`, but had lots of problems (related to MEF vs service locator), and when I stopped using the `AsyncLazy`, I forgot to get rid of this.